connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSinkTask.java

```diff
@@ -412,25 +434,23 @@ public void onPartitionsAssigned(Collection<TopicPartition> partitions) {
             // If we paused everything for redelivery (which is no longer relevant since we discarded the data), make
             // sure anything we paused that the task didn't request to be paused *and* which we still own is resumed.
             // Also make sure our tracking of paused partitions is updated to remove any partitions we no longer own.
-            if (pausedForRedelivery) {
-                pausedForRedelivery = false;
-
-                Set<TopicPartition> assigned = new HashSet<>(partitions);
-                Set<TopicPartition> taskPaused = context.pausedPartitions();
-
-                for (TopicPartition tp : partitions) {
-                    if (!taskPaused.contains(tp))
-                        consumer.resume(singleton(tp));
-                }
-
-                Iterator<TopicPartition> tpIter = taskPaused.iterator();
-                while (tpIter.hasNext()) {
-                    TopicPartition tp = tpIter.next();
-                    if (assigned.contains(tp))
-                        tpIter.remove();
-                }
+            pausedForRedelivery = false;
+
+            Set<TopicPartition> assigned = new HashSet<>(partitions);
+            Set<TopicPartition> taskPaused = context.pausedPartitions();
+
+            Iterator<TopicPartition> tpIter = taskPaused.iterator();
```

Comment:
Since we now have the subsequent check for `isPaused` (possibly soon to be `shouldPause` or similar), shouldn't we skip this logic if `isPaused()` is true? Seems like this should be in the `else` block instead of always run, only to sometimes follow up with pausing everything.